cmd/identity/keys.go

```diff
@@ -49,17 +68,142 @@ func cmdKeyGenerate(cmd *cobra.Command, args []string) (err error) {
 	counter := new(uint32)
 	return identity.GenerateKeys(ctx, uint16(keyCfg.MinDifficulty), keyCfg.Concurrency,
 		func(k *ecdsa.PrivateKey, id storj.NodeID) (done bool, err error) {
-			data, err := x509.MarshalECPrivateKey(k)
-			if err != nil {
-				return false, err
-			}
 			difficulty, err := id.Difficulty()
 			if err != nil {
 				return false, err
 			}
-			filename := fmt.Sprintf("gen-%02d-%d.key", difficulty, atomic.AddUint32(counter, 1))
-			fmt.Println("writing", filename)
-			err = ioutil.WriteFile(filepath.Join(keyCfg.OutputDir, filename), data, 0600)
+
+			genName := fmt.Sprintf("gen-%02d-%d.tar", difficulty, atomic.AddUint32(counter, 1))
+			dID := deferredIdentity{
+				tarPath: filepath.Join(keyCfg.OutputDir, genName),
+				key:     k,
+				id:      id,
+
+				tarData: new(bytes.Buffer),
+
+				errs: new(errs.Group),
+			}
+
+			go dID.build().writeTar(keyCfg.OutputDir).done()
```

Comment:
Not sure if this is a good idea... it's late and I was (am) tired.

Q's:
1. Separate go routine?
1. Chaining shenanigans?
1. Error shenanigans (and checking the error)?

Reply:
Yeah, no need for the goroutine.